diff --git a/node_modules/co-body/lib/json.js b/node_modules/co-body/lib/json.js
index 7bc8f12..2872133 100644
--- a/node_modules/co-body/lib/json.js
+++ b/node_modules/co-body/lib/json.js
@@ -38,9 +38,7 @@ module.exports = function(req, opts){
   // raw-body returns a promise when no callback is specified
   return Promise.resolve()
     .then(function() {
-      return raw(inflate(req), opts);
-    })
-    .then(function(str) {
+      const str = req.body;
       try {
         var parsed = parse(str);
         return opts.returnRawBody ? { parsed: parsed, raw: str } : parsed;
